extensions/mysql_extension/mysql/stream_push_handler.go

```diff
@@ -0,0 +1,84 @@
+package mysql
+
+import (
+	"bytes"
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"github.com/wal-g/wal-g/internal"
+	"github.com/wal-g/wal-g/internal/tracelog"
+	"io"
+	"os"
+	"strings"
+	"time"
+)
+
+func HandleStreamPush(uploader *Uploader, backupName string) {
+	fmt.Println("HIPE")
+	uploader.UploadingFolder = uploader.UploadingFolder.GetSubFolder(internal.BaseBackupPath)
+	db, err := getMySQLConnection()
+	if err != nil {
+		tracelog.ErrorLogger.Fatalf("%+v\n", err)
+	}
+
+	if backupName == "" {
+		backupName = StreamPrefix + time.Now().UTC().Format("20060102T150405Z")
+	}
+	stat, _ := os.Stdin.Stat()
+	var stream io.Reader = os.Stdin
+	if (stat.Mode() & os.ModeCharDevice) == 0 {
+		tracelog.InfoLogger.Println("Data is piped from stdin")
+	} else {
+		tracelog.ErrorLogger.Println("WARNING: stdin is terminal: operating in test mode!")
```

Comment:
In case when stdin is attached to terminal we should operate in test mode of fail. Current behavior is aimed at setting up things. But failing is acceptable too.